cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -562,24 +567,47 @@ open class Node @Inject constructor(private val project: Project) {
     /**
      * Installs the Dockerized configuration file to the root directory and detokenises it.
      */
-    internal fun installDockerConfig(defaultSsh: Int) {
+    internal fun installDockerConfig(defaultSSHPort: Int) {
         configureProperties()
+
         if (!config.hasPath("sshd.port")) {
-            sshdPort(defaultSsh)
+            sshdPort(defaultSSHPort)
         }
+
         val configDefaults = ConfigFactory.empty()
-            .withValue("dataSourceProperties.dataSource.url", ConfigValueFactory.fromAnyRef("jdbc:h2:file:./persistence/persistence;DB_CLOSE_ON_EXIT=FALSE;WRITE_DELAY=0;LOCK_TIMEOUT=10000"))
+                .withValue("dataSourceProperties.dataSource.url", ConfigValueFactory.fromAnyRef("jdbc:h2:file:./persistence/persistence;DB_CLOSE_ON_EXIT=FALSE;WRITE_DELAY=0;LOCK_TIMEOUT=10000"))
         val dockerConf = config
-            .withValue("p2pAddress", ConfigValueFactory.fromAnyRef("$containerName:$p2pPort"))
-            .withValue("rpcSettings.address", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.port}"))
-            .withValue("rpcSettings.adminAddress", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.adminPort}"))
-            .withValue("detectPublicIp", ConfigValueFactory.fromAnyRef(false))
-            .withFallback(configDefaults)
+                .withValue("p2pAddress", ConfigValueFactory.fromAnyRef("$containerName:$p2pPort"))
+                .withValue("rpcSettings.address", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.port}"))
+                .withValue("rpcSettings.adminAddress", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.adminPort}"))
+                .withValue("detectPublicIp", ConfigValueFactory.fromAnyRef(false))
+                .withFallback(configDefaults)
 
         config = dockerConf
         createNodeAndWebServerConfigFiles(config)
     }
 
+     /**
+     * Installs the default database (part of EG-117 Epic)
+     */
+    internal fun installDefaultDatabaseConfig(dbConfig: Config) {
+        val dockerConf = dbConfig.withFallback(config).resolve()
+        config = dockerConf
+        updateNodeConfigFile(config)
+    }
+
+    private fun updateNodeConfigFile(config: Config) {
+        val options = ConfigRenderOptions
+                .defaults()
+                .setOriginComments(false)
+                .setComments(false)
+                .setFormatted(true)
+                .setJson(false)
+        val configFileText = createNodeConfig(config).root().render(options).split("\n").toList()
+        val nodeConfFile = File(nodeDir, "node.conf")
+        Files.write(nodeConfFile.toPath(), configFileText)
```

Comment:
Do we need to specify `REPLACE_EXISTING` here as an option? We _are_ apparently _updating_ the configuration.

Reply:
We are re-writing/updating the same configuration file. Was thinking that option is used only when we copy files

Reply:
We're writing bytes to a file, and the Operating System wants us to tell it what to do if there is already a file with the same name on the filing system: Append, Replace or Abort?

Reply:
The default of Files.write is to create a new file or overwrites an existing file

Reply:
Fair enough.